### ISSUE TITLE
feat: provide a user friendly GUI

### ISSUE THREAD

**Issue by kemingy** [tensorchord/envd#1180]
## Description

`envd` CLI is suitable for software engineers. But ML is also widely used by people who don't have any CS background. A friendly GUI like GitHub Desktop (may not be a good example) can improve the user experience and attract more people to our community.

Requirements:
* need to include the dependencies like docker/podman
* can launch a jupyter notebook/VSCode

Framework candidates:
* [tauri](https://tauri.app/)
* [Electron](https://www.electronjs.org/)

---

**Message from the maintainers**:

Love this enhancement proposal? Give it a 👍. We prioritise the proposals with the most 👍.

**Comment by aseaday** [comment]
Is it a SoC Project?

**Comment by kemingy** [comment]
> Is it a SoC Project?

Yep. I think it's also open for everyone to take. (PS: it may take a lot of time)

**Comment by xieydd** [comment]
Maybe need some data to support opinion, i rarely see anyone using Docker Desktop.

**Comment by gaocegege** [comment]
Actually, you have to use docker desktop if you use macOS.

**Comment by xieydd** [comment]
I see.
